src/main/java/com/github/manolo8/darkbot/core/api/util/DataReader.java

```diff
@@ -0,0 +1,49 @@
+package com.github.manolo8.darkbot.core.api.util;
+
+import java.nio.ByteBuffer;
+
+public interface DataReader extends AutoCloseable {
+
+    int MAX_CHUNK_SIZE = 2048;
+
+    ByteBuffer getByteBuffer();
```

Comment:
Is this really required? i think the concept of data reader doesn't need to forcibly be backed by a byte buffer

the AbstractDataReader can add it as an abstract method and it should all work the same